nni/algorithms/compression/v2/pytorch/base/scheduler.py

```diff
@@ -0,0 +1,89 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from typing import List, Dict, Tuple
+
+from torch.nn import Module
+from torch.tensor import Tensor
+
+
+class BasePruningScheduler:
+    def generate_task(self) -> Tuple[int, Module, List[Dict], Dict[str, Dict[str, Tensor]]]:
+        """
+        Returns
+        -------
+        Tuple[int, Module, List[Dict], Dict[str, Dict[str, Tensor]]]
+            Return new task id, model under pruning, config list used in this task and pre-masks.
```

Comment:
A little bit concerned about the memory consumption. 
Do we need to deepcopy the module? if so, we may consume twice as much memory ? Besides, if we need save the dependent checkpoint for each task, it will consume more memory?

Reply:
Yes, we now use the disk save/load model to interact, but because pruner will reference the previous wrapped pruned model until the next model pass in, we will consume twice memory...
I will check the memory usage and fix this problem.

Reply:
or maybe the `task` is not a good implement, the logic is too complex, a specific scheduler for each scheduling algorithm may be better...

Reply:
add gc logic and test with psutil, the memory usage is stable